tests/machine/test_node.py

```
@@ -1,41 +1,265 @@
+                for (hn_src, port_src), (hn_dest, port_dest), num_bytes, entries in packets_expected
+            ) == self.packets_sent
+
+    return FakeSendPacket()
+
+
+def test_send_packet_address(endpoint):
+    with patch.object(endpoint, "_send_packet") as mock:
+        with endpoint.bind(9887) as socket:
+            socket.send(("172.99.80.23", 80), 45666)
+        with endpoint.bind(53) as socket:
+            socket.send(("8.8.8.8", 53), 652, Payload({PayloadDictionaryType.CONTENT: "google.ca"}))
+        mock.assert_has_calls(
+            [
+                call(Packet(Location(None, 9887), Location("172.99.80.23", 80), 45666, Payload({}))),
+                call(
+                    Packet(
+                        Location(None, 53),
+                        Location("8.8.8.8", 53),
+                        652,
+                        Payload({PayloadDictionaryType.CONTENT: "google.ca"})
+                    )
+                )
+            ]
+        )
+
+
+def test_resolve_destination_address(endpoint):
+    for s in ["8.8.8.8", "192.168.1.8"]:
+        assert endpoint.resolve_name(as_hostname(s)) == as_address(s)
+
+
+def test_send_packet_hostname(endpoint):
+    with patch.object(endpoint, "resolve_name", return_value="172.99.0.2"), \
+            patch.object(endpoint, "_send_packet") as mock:
+        with endpoint.bind(9887) as socket:
+            socket.send(("google.ca", 443), 3398)
+        mock.assert_called_once_with(Packet(Location(None, 9887), Location("172.99.0.2", 443), 3398, Payload({})))
+
+
+@contextmanager
+def run_simulation_receiving(socket, delay_recv, expected_end_time):
+    packet = Packet(
+        Location("192.168.2.89", 9887),
+        Location("172.99.0.2", 443),
+        12345,
+        Payload({PayloadDictionaryType.CONTENT: "Hello recv!"})
+    )
+
+    def receiving_on_endpoint():
+        pkt_received = socket.recv()
+        assert pkt_received == packet
```

For methods like these I think it is better to throw an exception and use `pytest.raises` to check for it, since the tests will pass in the case where this method is never called

---

Fixed in 1c040b6